### DISCUSSION TITLE
How to get the overrides parameter to work with useContractReads ?

### DISCUSSION THREAD

**Discussion by Fabien-Morrow** [wevm/wagmi#1069] [Q&A]
I have a smartcontract deployed on Goerli at [0x77a286b658191c9A8504D9e8E5c7B6381caA9e72](https://goerli.etherscan.io/address/0x77a286b658191c9A8504D9e8E5c7B6381caA9e72). It uses @openzeppelin/contracts/access/AccessControl.sol, allowing granting roles to some addresses, and baseUri() is allowed to be read only by USER_ROLE. The creator is 0x19D67dBE3018E2565A7b17Fe7F673770BC95a3FF, and so he has the USER_ROLE thanks to the constructor.

The contract is verified, you can check the code it's pretty straightforward.

I'm trying to read baseUri() using wagmi hook "useContractReads", overriding msg.sender with creator address and it fails. My code is :

mwe.js
```js
import Dummy from "../constants/Dummy.json";
import { useContractReads } from "wagmi";

export default function Mwe() {
  const { data1, error } = useContractReads({
    contracts: [
      {
        addressOrName: "0x77a286b658191c9A8504D9e8E5c7B6381caA9e72",
        contractInterface: Dummy.abi,
        functionName: "baseUri",
      },
    ],
    overrides: { from: "0x19D67dBE3018E2565A7b17Fe7F673770BC95a3FF" },
  });
  console.log("data1", data1);

  return <div>test</div>;
}
```
and It throws :
```
wagmi-core.esm.js?19bf:177 Contract method reverted with an error.

Config:
{
  "addressOrName": "0x77a286b658191c9A8504D9e8E5c7B6381caA9e72",
  "contractInterface": "...",
  "functionName": "baseUri",
  "chainId": 5
}

Details: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="baseUri()", errorArgs=["AccessControl: account 0xca11bde05977b3631167028862be2a173976ca11 is missing role 0x14823911f2da1b49f045a0929a60b8c1f2a7fc8c06c7284ca3e8ab4e193a08c8"], errorName="Error", errorSignature="Error(string)", reason="AccessControl: account 0xca11bde05977b3631167028862be2a173976ca11 is missing role 0x14823911f2da1b49f045a0929a60b8c1f2a7fc8c06c7284ca3e8ab4e193a08c8", code=CALL_EXCEPTION, version=abi/5.5.0)
```
My wagmi provider is
```js
alchemyProvider({ apiKey: process.env.GOERLI_RPC_KEY })
```
Well it seems it tries to read my contract with another address (provider by Alchemy I suppose) and obviously it fails ! Where is the bug in my code, or is there a bug in wagmi ?

**Comment by vinayprolitus** [comment]
I want to achieve this too. Want to override sender address.
